src/legacy/core_plugins/kbn_vislib_vis_types/public/controls/point_series/value_axes.js

```diff
@@ -1,205 +0,0 @@
-        }
-      };
-
-      $scope.updateAxisName = function (axis) {
-        const axisName = _.capitalize(axis.position) + 'Axis-';
-        axis.name = axisName + $scope.editorState.params.valueAxes.reduce((value, axis) => {
-          if (axis.name.substr(0, axisName.length) === axisName) {
-            const num = parseInt(axis.name.substr(axisName.length));
-            if (num >= value) value = num + 1;
-          }
-          return value;
-        }, 1);
-      };
-
-      const lastCustomLabels = {};
-      // We track these so we can know when the agg is changed
-      let lastMatchingSeriesAggType = '';
-      let lastMatchingSeriesAggField = '';
-      $scope.updateAxisTitle = function () {
-        $scope.editorState.params.valueAxes.forEach((axis, axisNumber) => {
-          let newCustomLabel = '';
-          const isFirst = axisNumber === 0;
-          const matchingSeries = [];
-          $scope.editorState.params.seriesParams.forEach((series, i) => {
-            const isMatchingSeries = (isFirst && !series.valueAxis) || (series.valueAxis === axis.id);
-            if (isMatchingSeries) {
-              let seriesNumber = 0;
-              $scope.editorState.aggs.forEach(agg => {
-                if (agg.schema.name === 'metric') {
-                  if (seriesNumber === i) matchingSeries.push(agg);
-                  seriesNumber++;
-                }
-              });
-            }
-          });
-
-          if (matchingSeries.length === 1) {
-            newCustomLabel = matchingSeries[0].makeLabel();
-          }
-
-          const matchingSeriesAggType = _.get(matchingSeries, '[0]type.name', '');
-          const matchingSeriesAggField = _.get(matchingSeries, '[0]params.field.name', '');
-
-          if (lastCustomLabels[axis.id] !== newCustomLabel && newCustomLabel !== '') {
-            const isFirstRender = Object.keys(lastCustomLabels).length === 0;
-            const aggTypeIsChanged = lastMatchingSeriesAggType !== matchingSeriesAggType;
-            const aggFieldIsChanged = lastMatchingSeriesAggField !== matchingSeriesAggField;
-            const aggIsChanged = aggTypeIsChanged || aggFieldIsChanged;
-            const axisTitleIsEmpty = axis.title.text === '';
-            const lastCustomLabelMatchesAxisTitle = lastCustomLabels[axis.id] === axis.title.text;
-
-            if (!isFirstRender && (aggIsChanged || axisTitleIsEmpty || lastCustomLabelMatchesAxisTitle)) {
-              axis.title.text = newCustomLabel; // Override axis title with new custom label
-            }
-
-            lastCustomLabels[axis.id] = newCustomLabel;
-          }
-
-          lastMatchingSeriesAggType = matchingSeriesAggType;
-          lastMatchingSeriesAggField = matchingSeriesAggField;
-        });
-      };
-
-      $scope.$watch(() => {
-        return $scope.editorState.aggs.map(agg => {
-          return safeMakeLabel(agg);
-        }).join();
-      }, () => {
-        $scope.updateAxisTitle();
```

Comment:
note: moved to https://github.com/elastic/kibana/pull/43772/files?file-filters%5B%5D=.tsx&hide-deleted-files=true#diff-af178345ae0266d7293defbfeef8b57bR192